### PR TITLE
Cleanup and fix tasks

### PR DESCRIPTION
**Pull request by drmarkwslater**
This PR fixes Tasks at least from the Core point of view. Everything now seems to work with branch both locally and with Dirac.

It's probably worth mentioning that the vast majority (all?) of this is cleanup. The main fix that was needed was sorting out makeGangaListByRef in PR #345 .

### CODE REVIEW

File: python/Ganga/new_tests/GPI/Tasks/TestCoreTasks.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+        from Ganga.GPIDev.Base.Proxy import isType
+        from Ganga.GPI import tasks, CoreTask, CoreTransform, Executable, GenericSplitter
+        t = CoreTask()
+
+        trf = CoreTransform()
+        trf.application = Executable()
+        trf.unit_splitter = GenericSplitter()
+        trf.unit_splitter.attribute = "application.args"
+        trf.unit_splitter.values = [ 'arg 1', 'arg 2', 'arg 3' ]
```
Comment (milliams) [inline python/Ganga/new_tests/GPI/Tasks/TestCoreTasks.py]: PEP8 formatting. [There shouldn't be spaces inside brackets](https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements). On a few other lines in the tests too.

**Comment by rob-c** [comment]
Go for merge from me :+1:

**Comment by drmarkwslater** [comment]
retest this please

File: python/Ganga/new_tests/GPI/Tasks/TestCoreTasks.py (hunk 1 above)
Comment (drmarkwslater) [inline python/Ganga/new_tests/GPI/Tasks/TestCoreTasks.py]: I always miss that :) fixed!

**Comment by rob-c** [comment]
I'm surprised that we're seeing tests stalling on this branch as it doesn't touch anything in the Core other than Tasks does it?

**Comment by drmarkwslater** [comment]
I don't what happened to the 642-644 tests as they all took ages and failed. But the last test for this worked fine and the current one seems to be OK as well. I'll probably give it one last go just to be sure the merge if all is well.

**Comment by drmarkwslater** [comment]
retest this please

**Comment by drmarkwslater** [comment]
retest this please

**Comment by rob-c** [comment]
Could this PR be effecting the order in which modules are loaded?

**Comment by drmarkwslater** [comment]
retest this please

**Comment by drmarkwslater** [comment]
retest this please

**Comment by drmarkwslater** [comment]
retest this please

**Comment by drmarkwslater** [comment]
retest this please

**Comment by drmarkwslater** [comment]
retest this please

**Comment by drmarkwslater** [comment]
It looks like this is now working with the change to the `hostname()` function. Any problems with a merge?

**Comment by rob-c** [comment]
Glad to see this was understood, I've seen this before but have thought I'm going crazy. Look forward to seeing if this gets rid of oddities in other branches.

**Comment by rob-c** [comment]
Go for merge from me :+1:

File: python/Ganga/GPIDev/Lib/Tasks/CoreUnit.py [hunk 2]
```diff
@@ -32,7 +33,7 @@ def createNewJob(self):
             if (f == "postprocessors" and len(getattr(self, f).process_objects) > 0):
                 j.postprocessors = copy.deepcopy(addProxy(self).postprocessors)
             elif (f != "postprocessors" and getattr(self, f)):
-                setattr(j, f, copy.deepcopy(getattr(self, f)))
+                setattr(stripProxy(j), f, copy.deepcopy(getattr(self, f)))
```
Comment (milliams) [inline python/Ganga/GPIDev/Lib/Tasks/CoreUnit.py]: I think that `j` should be a raw job here? `makeRegisteredJob()` certainly returns one.

Comment (drmarkwslater) [inline python/Ganga/GPIDev/Lib/Tasks/CoreUnit.py]: I hadn't seen the change to `makeRegisteredJob`. Fixed.

**Comment by drmarkwslater** [comment]
Any other issues with merging this?

**Comment by milliams** [comment]
Once the tests pass :+1:

**Pull request merged**